resources/todomvc/architecture-examples/svelte/src/Item.svelte

```diff
@@ -0,0 +1,51 @@
+<script>
+    import { createEventDispatcher, tick } from 'svelte';
+
+    const dispatch = createEventDispatcher();
+
+    function removeItem() {
+        dispatch('removeItem', { index });
+    }
+
+    function startEdit() {
+        dispatch('startEdit', { index });
+    }
+
+    function handleEdit(event) {
+        dispatch('handleEdit', { key: event.key, target: event.target})
+    }
+
+    function updateItem(event) {
+        dispatch('updateItem', { text: event.target.value, index });
+    }
+
+    function toggleItem(event) {
+        dispatch('toggleItem', { index, checked: event.target.checked });
+    }
+
+    async function focusInput(element) {
+		await tick();
+		element.focus();
+	}
+
+    export let item;
+    export let editing;
```

Comment:
It's not clear to me why the `editing` state is handled outside of this component. I would have used a boolean handled at the component level, and not exposed it to the parent. The `index` wouldn't be necessary.

Note also that with the current implementation all items are rerendered if the "editing" state changes for one (because `editing` changes for all). This sounds suboptimal.